### DISCUSSION TITLE
Pico not recognized after updating micropython to 1.24.0

### DISCUSSION THREAD

**Discussion by jmw0149** [micropython/micropython#16196] [RP2040 / Pico]
I bought a batch of picos in Nov 2021. Been working with 2 of them since with no problem. Decided to update the micropython firmware. The units then lost connection to the computer with no COM port available. They will now mount in boot mode and Thonny will download the file to the pico then ends with a message that a com port cannot be found as expected when the update completes. The computer does not detect a com port when reinserted; widows gives a device connection failure message. Tested with both Windows 11 Thonny, Raspberry Pi Thonny and manual drag/drop. Neither computer can find a pico com port after updating. Tried 2 other unused picos in the same batch and had the same problem. Tried installing USB drivers with zadig; that did not correct the problem but did display an error message for the installed driver: "Device Descriptor Request Failed". Located ver 1.23.0 archive and installed by drag & drop. The com ports now function as expected on all units.

**Comment by scruss** [comment]
This is an odd one: I have Pico boards of the same age, and have updated them to v1.24.0 without any issues.

Was it the same download of the uf2 in all cases? Maybe it didn't download properly if so.

[RPI_PICO-20241025-v1.24.0.uf2](https://micropython.org/resources/firmware/RPI_PICO-20241025-v1.24.0.uf2): 666624 bytes. md5sum:
```
bf5ea2d72677a1142c412014f034831e  RPI_PICO-20241025-v1.24.0.uf2
```
Image also verifies using `picotool verify RPI_PICO-20241025-v1.24.0.uf2` against a flashed Pico board in bootloader mode.

**Reply by jmw0149** [reply]
Downloaded multiple times. Doesn't matter, it always the same error with the latest version.  I spent several hours doing multiple tests of  various versions. Learned much more than I ever wanted to about flashing a pico and com ports.

**Comment by mendenm** [comment]
Any chance you downloaded the Pico-W firmware for a plain Pico?  That would do it, and it is an easy mistake to make.

**Reply by jmw0149** [reply]
Nope.

**Comment by TurBoss** [comment]
hello, I'm having the same issue in all of the pico w  I have

only works for 1.23 
not works for 1.24 and 1.25

it only works if is empty, after flashing nuke.uf2 to wipe the flash then laod micropython 1.24 or 1.25 but as soon I upload my code it stops showing same for MSC (built from source) as soon i upload the code it doesn't show any more

thanks

**Reply by scruss** [reply]
This is a different problem. MicroPython doesn't do MSC on the Raspberry Pi Pico W.

**Reply by TurBoss** [reply]
also no uart?

thanks

**Reply by scruss** [reply]
does the stock [PICO_W-20241025-v1.24.0](https://micropython.org/resources/firmware/RPI_PICO_W-20241025-v1.24.0.uf2) firmware work?

**Reply by robert-hh** [reply]
Do you connect the Pico directly to the PC or though a USB hub. I have sometimes problems that board are not detected when being connected at a USB hub.

**Comment by jmw0149** [comment]
I'll write this one off as a learning experience. I can reliably switch between builds and get it to disappear or work normally. These chips are now destined for some seasonal LED displays where the silicon in the chip will turn back into sand before it'll be upgraded.

**Reply by scruss** [reply]
That's unfortunate. If you're ever able to get to the bottom of this and document it for others here, it would be very helpful.
